apps/04TowersOfHanoi.js

```diff
@@ -16,25 +16,75 @@ function printStacks() {
     console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-    // Your code here
+function movePiece(startStack, endStack) {
+
+  var piece = stacks[startStack].pop()
+  stacks[endStack].push(piece);
 
 }
 
-function isLegal() {
-    // Your code here
+function isLegal(startStack, endStack) {
 
-}
+
+  var startStackArray = stacks[startStack];
+
+  var endStackArray = stacks[endStack];
+
+  if ( startStackArray.length === 0 )   {
+
+    return false;
+
+  }
+
+  else if (endStackArray.length === 0)  {
+
+      return true;
+
+  }
+
+  else {
+
+      return startStackArray[startStackArray.length - 1] < endStackArray[endStackArray.length - 1];
+
+  }
+
+  }
 
 function checkForWin() {
     // Your code here
+    if (stacks.c.length === 4)   {
```

Comment:
this looks a lot like the next `if` block, could we combine them somehow? Maybe with an OR ?